docker-app/qfieldcloud/core/middleware/client_type.py

```diff
@@ -0,0 +1,29 @@
+from qfieldcloud.authentication.models import AuthToken
+
+
+class ClientTypeMiddleware:
+    """Client type middleware.
+
+    This middleware detects the client type based on the user agent
+    and sets it on the session object.
+
+    This client_type is required by Views later on during the QField/Sync process.
+
+    Typically used for users connected with SSO, which don't have any `AuthToken`
+    attached to the session, but future requests still need to know the client type.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
```

Comment:
A type hint like `request: HttpRequest` could be used here (`from django.http import HttpRequest`).